src/ResourceManager/Profile/Commands.Profile.Test/LongRunningCmdletTests.cs

```diff
@@ -0,0 +1,226 @@
+        static readonly ErrorRecord Error = new ErrorRecord(new InvalidOperationException("invalid operation"), "12345", ErrorCategory.InvalidOperation, new object());
+        static readonly object Output = new TestCmdletOutput();
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void CanReceiveAllStreams()
+        {
+            Mock<ICommandRuntime> mockRuntime;
+            var cmdlet = SetupCmdlet(false, false, out mockRuntime);
+            var job = cmdlet.ExecuteAsJob("Test Job") as AzureLongRunningJob<AzureStreamTestCmdlet>;
+            int times = 0;
+            while (times++ < 20 && job.StatusMessage != "Completed")
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(1));
+            }
+
+            ValidateCompletedCmdlet(job);
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void CanSupportShouldProcess()
+        {
+            Mock<ICommandRuntime> mockRuntime;
+            var cmdlet = SetupCmdlet(true, false, out mockRuntime);
+            var job = cmdlet.ExecuteAsJob("Test Job") as AzureLongRunningJob<AzureStreamTestCmdlet>;
+            int times = 0;
+            while (times++ < 20 && job.StatusMessage != "Completed")
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(1));
+            }
+
+            ValidateCompletedCmdlet(job);
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void CanSupportShouldContinue()
+        {
+            Mock<ICommandRuntime> mockRuntime;
+            var cmdlet = SetupCmdlet(true, true, out mockRuntime);
+            var job = cmdlet.ExecuteAsJob("Test Job") as AzureLongRunningJob<AzureStreamTestCmdlet>;
+            int times = 0;
+            while (times++ < 20 && job.StatusMessage != "Completed")
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(1));
+                if (job.StatusMessage == "Blocked")
+                {
+                    job.Start();
+                }
+            }
+
+            ValidateCompletedCmdlet(job);
+            mockRuntime.Verify(m => m.ShouldContinue(It.IsAny<string>(), It.IsAny<string>()), Times.Exactly(1));
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void CanHandleCmdletException()
+        {
+            Mock<ICommandRuntime> mockRuntime;
+            var cmdlet = SetupCmdlet(false, false, out mockRuntime);
+            cmdlet.Fail = true;
+            var job = cmdlet.ExecuteAsJob("Test Job") as AzureLongRunningJob<AzureStreamTestCmdlet>;
+            int times = 0;
+            while (times++ < 20 && job.StatusMessage != "Completed" && job.StatusMessage != "Failed")
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(1));
+            }
+
+            Assert.Equal("Failed", job.StatusMessage);
+            Assert.Equal(2, job.Error.Count);
```

Comment:
@markcowl what do the streams look like for a non-completed job? Are they something that we should validate?

Reply:
We do have tests of stopped job, but it is difficult for a mocked test to reliably check for the state of the fields in between.